pydatastructs/linear_data_structures/algorithms.py

```diff
@@ -1740,3 +1741,114 @@ def jump_search(array, value, **kwargs):
         prev += 1
 
     return None
+
+def intro_sort(array, **kwargs) -> Array:
+    """
+    Performs intro sort on the given array.
+
+    Parameters
+    ==========
+
+    array: Array
+        The array which is to be sorted.
+    start: int
+        The starting index of the portion
+        which is to be sorted.
+        Optional, by default 0
+    end: int
+        The ending index of the portion which
+        is to be sorted.
+        Optional, by default the index
+        of the last position filled.
+    maxdepth: Enables the user to define the maximum
+        recursion depth, takes value 2*log(length(A))
+        by default (ref: Wikipedia[1]).
+    ins_threshold: Threshold under which insertion
+        sort has to be performed, default value is
+        16 (ref: Wikipedia[1]).
+    backend: pydatastructs.Backend
+        The backend to be used.
+        Optional, by default, the best available
+        backend is used.
+
+    Returns
+    =======
+
+    output: Array
+        The sorted array.
+
+    Examples
+    ========
+
+    >>> from pydatastructs import OneDimensionalArray as ODA, intro_sort
+    >>> arr = ODA(int, [5, 78, 1, 0])
+    >>> out = intro_sort(arr)
+    >>> str(out)
+    '[0, 1, 5, 78]'
+    >>> arr = ODA(int, [21, 37, 5])
+    >>> out = intro_sort(arr)
+    >>> str(out)
+    '[5, 21, 37]'
+
+    Note
+    ====
+
+    This function does not support custom comparators as
+    is the case with other sorting functions in this file.
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Introsort
+    """
+    backend = kwargs.pop("backend", Backend.PYTHON)
+    if backend == Backend.CPP:
+        return _algorithms.intro_sort(array, **kwargs)
+
+    # from pydatastructs import Stack
+    # Always sorts in increasing order
```

Comment:
I think this is because of heapsort's limitation, right? Please mention here about this.

Reply:
Yes, that is because of the current heapsort function's limitation.

Reply:
Should I write that as a comment in the code?

Reply:
Changes made.